winch/codegen/src/visitor.rs

```diff
@@ -2285,6 +2293,54 @@ where
         )
     }
 
+    fn visit_i32_atomic_rmw8_add_u(&mut self, arg: MemArg) -> Self::Output {
+        self.atomic_rmw(
+            &arg,
+            RmwOp::Add,
+            OperandSize::S8,
+            Some(ExtendKind::I32Extend8S),
+        )
+    }
+
+    fn visit_i32_atomic_rmw16_add_u(&mut self, arg: MemArg) -> Self::Output {
+        self.atomic_rmw(
+            &arg,
+            RmwOp::Add,
+            OperandSize::S16,
+            Some(ExtendKind::I32Extend16S),
```

Comment:
The extension kinds used in this case are signed, however the [spec states zero extension instead](https://github.com/WebAssembly/threads/blob/main/proposals/threads/Overview.md#read-modify-write). I believe we need to augment `ExtendKind` to account for the missing unsigned extensions.

Reply:
that's because https://github.com/bytecodealliance/wasmtime/blob/main/winch/codegen/src/isa/x64/asm.rs#L395-L401 takes `ExtendKind`, even though `movzx` zero-extend. Can I do that in a followup, if thats ok? i'll add variants for unsigned conversions, and update any affected code.

Reply:
I've extended `ExtendKind` here https://github.com/bytecodealliance/wasmtime/pull/10012 to include unsigned extends.